include/kokkos-utils/impl/type_list.hpp

```diff
@@ -154,26 +160,47 @@ inline constexpr bool type_list_all_v = type_list_all<UnaryPred, TypeList>::valu
  *       See also https://github.com/kokkos/kokkos/blob/db736f280b09ae5acaedab3b3ad6bc7d741e92bc/core/src/impl/Kokkos_Utilities.hpp#L171-L176.
  */
 ///@{
-template <template <typename> class UnaryPred, class List>
+template <template <typename> typename UnaryPred, typename...>
 struct type_list_any;
 
+/// @name Specialization for a "direct" list of types.
+/// As of @c Cuda 12.8, using @c std::disjunction sometimes ends up erroring out with
+/// @code
+/// 'Kokkos::utils::impl::type_list_any<...>' has a base 'std::disjunction<...>' which has internal linkage [-Werror=subobject-linkage]
```

Comment:
It's a bit weird. Have you been able to see which "internally linked" variable it's complaining about? May be interesting to see if it's a question of includes. But I'm not sure.

Reply:
Clearly some `nvcc` specific stuff. Let's not waste time on this. BTW the error was 666 kilometers long.